app/Dockerfile

```diff
@@ -8,20 +8,17 @@ WORKDIR /usr/src/app
 ENV PYTHONDONTWRITEBYTECODE 1
 ENV PYTHONUNBUFFERED 1
 ENV PYTHONPYCACHEPREFIX=/root/.cache/pycache/
-ENV PIP_CACHE_DIR=/var/cache/buildkit/pip
-
-RUN mkdir -p $PIP_CACHE_DIR
-RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
 
 # install system dependencies
 RUN \
-  --mount=type=cache,target=/var/cache/apt,sharing=locked \
-  --mount=type=cache,target=/var/lib/apt,sharing=locked \
-  apt-get update \
-  && apt-get install --no-install-recommends -yqq \
-  netcat=1.10-46 \
-  gcc=4:10.2.1-1 \
-  graphviz=2.42.2-5
+  --mount=type=cache,target=/var/cache/apk \
+  --mount=type=cache,target=/etc/apk/cache \
+  apk add \
+  graphviz=9.0.0-r0
+
```

Comment:
Hey @fyliu while trying to run `scripts/buildrun.sh` encountered an error with the service `web` as it failed to execute, maybe there's an issue with the `graphic version`. Below is the error block

```bash

=> ERROR [web stage-0  3/10] RUN   --mount=type=cache,target=/var/cache/apk   --mount=type=cache,target=/etc/apk/cache   apk add   graphviz=9.0.0-r0    1.1s
------
 > [web stage-0  3/10] RUN   --mount=type=cache,target=/var/cache/apk   --mount=type=cache,target=/etc/apk/cache   apk add   graphviz=9.0.0-r0:
0.130 fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/main/aarch64/APKINDEX.tar.gz
0.517 fetch https://dl-cdn.alpinelinux.org/alpine/v3.20/community/aarch64/APKINDEX.tar.gz
1.052 ERROR: unable to select packages:
1.053   graphviz-9.0.0-r1:
1.053     breaks: world[graphviz=9.0.0-r0]
------
failed to solve: process "/bin/sh -c apk add   graphviz=9.0.0-r0" did not complete successfully: exit code: 1

```

Reply:
I found out that graphviz only exists in a single released version at any point in time. So the previous release is purged when a new one is uploaded. To work around this so future docker builds don't fail this way, I'm setting a minimum version to the minor version. It will still fail if that project decides to release v9.1, but that's possibly a breaking change and we should re-test the package at that point. See the [official documentation](https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper#Advanced_APK_Usage) for syntax.